src/ipc.ts

```diff
@@ -0,0 +1,36 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) Jupyter Development Team.
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+
+export
+namespace JupyterAppChannels {
+    
+    /**
+     * IPC channel for render process signalling
+     */
+    export
+    let RENDER_PROCESS_READY = 'render-ready';
+
+    /**
+     * IPC channel for sending jupyter server data
+     */
+    export
+    let SERVER_DATA = 'server-data';
+}
+
+export
+namespace JupyterMenuChannels {
+
+    /**
+     * IPC channel for appending menus to the native menu bar
+     */
+    export
+    let MENU_ADD = 'menu-add';
+
+    /**
+     * IPC channel for recieveing click events
```

Comment:
Thanks for the feedback @willingc! I will be sure to put more care into my documentation in the future. 

Looks like there might be a problem with how this is running on your computer. I don't see the 'File' menu appearing, and the upper left corner says 'Electron' instead of 'JupyterLab'. Is the 'Electron' menu item clickable? Perhaps showing the contents of file menu?